ember-toucan-core/unpublished-development-types/index.d.ts

```diff
@@ -3,8 +3,11 @@
 
 import '@glint/environment-ember-loose';
 
+import type FieldComponent from '../src/components/form/field';
+
 declare module '@glint/environment-ember-loose/registry' {
   export default interface Registry /* extends EmberPageTitle, etc, other addon registries */ {
     // local entries
+    'Form::Field': typeof FieldComponent;
```

Comment:
I'm not quite sure _why_ I need to add this here, I just found by adding it here as well the glint error (above) was resolved and things were happy.  I can only _assume_ we need this as the `template-registry` above and the registry here are "disconnected" as-in they don't know about each other.

Reply:
yeah -- the declare module is what glint actually uses for lookup -- you don't want to ship a declare module, because that breaks TS consumers

Reply:
Do I need to add to both ^^ files as well when I add new components then?

Reply:
Yes, exactly!

The key here is `declare module '@glint/environment-ember-loose/registry' { ... }`. This is what makes things known for the (classic non-strict-mode) template. And it _must_ exist only in one place for any app. So the app has the declaration, and there it can add its own type definitions and/or add existing registries imported from glint-enabled addons . See https://typed-ember.gitbook.io/glint/using-glint/ember/using-addons#using-glint-enabled-addons

So this file is indeed only local to the addon. The `template-registry.ts` doesn't do _anything_ (has no side-effect), as it doesn't have that `declare module '@glint/environment-ember-loose/registry'`. So only when this is imported from the consuming app and added to _its_ `@glint/environment-ember-loose/registry` will this have any effect.

Reply:
> yeah -- the declare module is what glint actually uses for lookup -- you don't want to ship a declare module, because that breaks TS consumers

Perfect, thanks so much for the clarification! Makes sense!

Reply:
y'all are awesome, ty!

Reply:
> Do I need to add to both ^^ files as well when I add new components then?

@nicolechung I think it depends (based on my understanding of above):

- If your component is **only** going to be used externally, putting it in `template-registry` is just fine
- If your component is going to be used _internally_ to toucan-core, then you'll need it in `unpublished-development-types/index.d.ts` (say, as a building-block component for other components in toucan-core)
- If it's going to be used _both_ internally and externally then both places

Reply:
Thank you for the clarification!